sdk/storage/Azure.Storage.DataMovement/tests/Shared/StartTransferUploadDirectoryTestBase.cs

```diff
@@ -0,0 +1,484 @@
+            : base(async, mode)
+        { }
+
+        protected string GetNewObjectName(int? maxChars = 8)
+        {
+            string result = ClientBuilder.Recording.Random.NewGuid().ToString();
+            return maxChars < result.Length ? result.Substring(0, maxChars.Value) : result;
+        }
+
+        #region Service-Specific Implementations
+        /// <summary>
+        /// Gets a service-specific disposing container for use with tests in this class.
+        /// </summary>
+        /// <param name="service">Optionally specified service client to get container from.</param>
+        /// <param name="containerName">Optional container name specification.</param>
+        protected abstract Task<IDisposingContainer<TContainerClient>> GetDisposingContainerAsync(
+            TServiceClient service = default,
+            string containerName = default);
+
+        /// <summary>
+        /// Initializes data at the destination container.
+        /// </summary>
+        /// <returns></returns>
+        protected abstract Task InitializeDestinationDataAsync(
+            TContainerClient containerClient,
+            List<(string FilePath, long Size)> fileSizes,
+            CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Gets the specific storage resource from the given TObjectClient
+        /// e.g. ShareFileClient to a ShareFileStorageResource, BlockBlobClient to a BlockBlobStorageResource.
+        /// </summary>
+        /// <param name="objectClient">The object client to create the storage resource object.</param>
+        /// <returns></returns>
+        protected abstract StorageResourceContainer GetStorageResourceContainer(TContainerClient containerClient);
+
+        /// <summary>
+        /// Gets the appropriate delegate for listing through and validating the state of the destination container.
+        /// </summary>
+        /// <param name="containerClient"></param>
+        /// <param name="path"></param>
+        /// <returns></returns>
+        protected abstract TransferValidator.ListFilesAsync GetStorageResourceLister(TContainerClient containerClient);
+        #endregion
+
+        #region Test Helpers
+        private async Task SetupDirectoryAsync(
+            string directoryPath,
+            List<(string FilePath, long Size)> fileSizes,
+            CancellationToken cancellationToken)
+        {
+            foreach ((string filePath, long size) in fileSizes)
+            {
+                string currRelPath = "";
+                string[] pathSegments = filePath.Split('/', '\\');
+                if (pathSegments.Length < 1)
+                {
+                    continue;
+                }
+                foreach (string directoryName in pathSegments.Take(pathSegments.Length - 1))
+                {
+                    currRelPath = string.Join("/", currRelPath, directoryName).Trim('/');
```

Comment:
Does this need to be using the `Path.DirectorySeparatorChar` so it can use the platform's preferred separator?

https://learn.microsoft.com/en-us/dotnet/api/system.io.path.directoryseparatorchar?view=net-7.0